### PR TITLE
feat: Implement nsSphereAuthorityAuthorizationsList Swift FFI wrapper

### PR DESCRIPTION
**Pull request by bfollington**
Blindly copying the other examples, but it seems to work (both in tests and in Subconscious itself) 😬

### CODE REVIEW

File: swift/Tests/SwiftNoosphereTests/NoosphereTests.swift [hunk 1]
```diff
@@ -553,10 +553,10 @@ final class NoosphereTests: XCTestCase {
             print("Authorization:", authorization)
             
             ns_sphere_save(noosphere, sphere, nil, nil)
-            
-            nsSphereAuthorityEscalate(noosphere, sphere, sphere_mnemonic_ptr) {
-                (error, root_sphere) in
-                
+
+            nsSphereAuthorityAuthorizationsList(noosphere, sphere) {
+                (error, authorizations) in
```
Comment (jsantell) [inline swift/Tests/SwiftNoosphereTests/NoosphereTests.swift:558]: I believe this `authorizations` (an array of owned-strings) will need to be freed

File: swift/Tests/SwiftNoosphereTests/NoosphereTests.swift [hunk 2]
```diff
@@ -584,12 +584,48 @@ final class NoosphereTests: XCTestCase {
...
+                            ns_error_free(error)
+                            return
+                        }
+
+                        print("Authorization revoked!")
+
+                        ns_sphere_save(noosphere, root_sphere, nil, nil)
+
+                        nsSphereAuthorityAuthorizationsList(noosphere, root_sphere) {
+                            (error, authorizations) in
```
Comment (jsantell) [inline swift/Tests/SwiftNoosphereTests/NoosphereTests.swift:608]: Same, `authorizations` needs to be freed

**Review by jsantell — APPROVED** [review]
Fantastic! Some comments on freeing an authorizations list

File: swift/Sources/SwiftNoosphere/Noosphere.swift [hunk 3]
```diff
@@ -166,3 +166,22 @@ public func nsSphereAuthorityAuthorizationRevoke(_ noosphere: OpaquePointer!, _
         handler.contents(error)
     }
 }
+
+public typealias NsSphereAuthorityListAuthorizationsHandler = (OpaquePointer?, slice_boxed_char_ptr_t?) -> ()
```
Comment (subconscious-ops) [inline swift/Sources/SwiftNoosphere/Noosphere.swift:170]: Conventionally, verb should come _after_ `Authorizations` e.g., `NsSphereAuthorityAuthorizationsListHandler`

**Comment by cdata** [comment]
Please rebase on main to incorporate #514 (fixes CI browser test jobs).

**Comment by bfollington** [comment]
It has been done, everything is green ✅

**Review by jsantell — APPROVED** [review]
Great work @bfollington !

**Pull request merged**